commons/src/store/contacts.ts

```diff
@@ -1,50 +1,95 @@
 import { writable } from "svelte/store";
-import { fetchContactsByQuery } from "@commons";
+import {
+  fetchContacts,
+  fetchContactsByQuery,
+} from "@commons/connections/contacts";
 import type {
-  StoredContacts,
   Contact,
+  ContactEmail,
   ContactSearchQuery,
+  ContactsQuery,
 } from "@commons/types/Contacts";
 
-const contactsMap: Record<string, Contact[]> = {};
+let contactsMap: Record<string, Contact[]> = {};
+
+function filterContacts(contacts: Contact[]) {
+  return contacts
+    .filter(
+      (contact) =>
+        !!contact.given_name ||
+        !!contact.surname ||
+        (Array.isArray(contact.emails) && contact.emails.length > 0),
+    )
+    .map((contact) => {
+      // Ensure each contact has at least one "email" to load
+      if (!Array.isArray(contact.emails) || contact.emails.length === 0) {
+        contact.emails = [{ email: "" } as ContactEmail];
+      }
+
+      return contact;
+    });
+}
 
 function initializeContacts() {
   const { subscribe, set, update } = writable<Record<string, Contact[]>>({});
   return {
     subscribe,
-    addContacts: (incomingContacts: StoredContacts) => {
-      update((contacts) => {
-        contacts[incomingContacts.queryKey] = contacts[
-          incomingContacts.queryKey
-        ]
-          ? [...contacts[incomingContacts.queryKey], ...incomingContacts.data]
-          : incomingContacts.data;
-        return contacts;
-      });
+    addContacts: async (
+      query: ContactsQuery,
+      offset: number,
+      limit: number,
+    ) => {
+      const queryKey = JSON.stringify(query);
+      if (
+        !contactsMap[queryKey] &&
+        (query.component_id || query.access_token)
+      ) {
+        if (offset === 0) {
+          // Ensure the store is empty if our offset is 0
+          ContactStore.reset();
+        }
```

Comment:
You know this works great for `offset = 0` to avoid duplicates. But if someone had `offset = 1`, then `offset = 2` and then set it back to `offset = 1` - wouldn't we get duplicates in our store regardless?

Reply:
I think the logic hopes that the component handles this properly, but you're right that this isn't great. The logic has just been moved out of the fetch method.